lib/checkPullRequestReview.js

```diff
@@ -0,0 +1,290 @@
+  );
+
+  if (reviewerIsAssigned) {
+    // Unassign the reviewer if still assigned.
+    await context.github.issues.removeAssignees(
+      context.repo({
+        issue_number: pullRequest.number,
+        assignees: [reviewer],
+      })
+    );
+
+    await context.github.issues.createComment(
+      context.repo({
+        issue_number: pullRequest.number,
+        body: 'Unassigning @' + reviewer + ' since the PR is approved.',
+      })
+    );
+  }
+
+  // Check if pull request has been approved by all reviewers.
+  // The pull request data does not contain information if the pull request
+  // has been approved, so we are searching for an approved pull request
+  //  with the pull request number and in the appropriate repository. An
+  // empty array will be returned if the pull request is not approved by
+  // all reviewers.
+  const searchResult = await context.github.search.issuesAndPullRequests(
+    context.repo({
+      q:
+        'repo:' +
+        context.payload.repository.full_name +
+        ' review:approved ' +
+        pullRequest.number,
+    })
+  );
+
+  const searchData = searchResult.data.items;
+  const assignees = pullRequest.assignees.map((assignee) => assignee.login);
+  if (searchData.length === 0) {
+    // Pull request has not been approved by all reviewers, hence we need to
+    // assign the remaining reviewers.
+    const reviewersYetToReview = pullRequest.requested_reviewers.map(
+      (reviewer) => reviewer.login
+    );
+
+    const unassignedReviewers = reviewersYetToReview.filter(
+      (reviewer) => !assignees.includes(reviewer)
+    );
+    if (unassignedReviewers.length > 0) {
+      // Ping and assign unassigned reviewers.
+      const commentBody =
+        'Assigning @' +
+        unassignedReviewers.join(', @') +
+        ' for code owner reviews, Thanks!';
+
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body: commentBody,
+        })
+      );
+
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: unassignedReviewers,
+        })
+      );
+    }
+  } else {
+    // The pull request has been approved, and we need to add the LGTM label.
+    const labels = pullRequest.labels.map((label) => label.name);
+    if (labels.includes(LGTM_LABEL)) {
+      // Do nothing and end execution if label has already been added.
+      return;
+    }
+    await context.github.issues.addLabels(
+      context.repo({
+        issue_number: pullRequest.number,
+        labels: [LGTM_LABEL],
+      })
+    );
+
+    // The last reviewer got unassigned, but the pull request assignees list
+    // does not reflect that since we did not refetch the pull request data,
+    // hence if any other person is assigned to the pull request, they are
+    // likely the one to merge it.
+    console.log({assignees})
```

Comment:
We can remove the console.log, I think.

Also, we should add a comment before returning, asking the assignee for merging.

And why not just check membership of the assignees instead of comparing them to reviewer?

Reply:
> We can remove the console.log, I think.

Removed.

> And why not just check membership of the assignees instead of comparing them to reviewer?

I believe all reviewers should have merging rights, right?

> Also, we should add a comment before returning, asking the assignee for merging.

Comment add